lib/timers.js

```diff
@@ -418,6 +404,27 @@ function processImmediate() {
 }
 
 
+function tryOnImmediate(immediate, queue) {
+  var threw = true;
+  try {
+    immediate._onImmediate();
+    threw = false;
+  } finally {
+    if (threw === true) {
```

Comment:
If you do something like
```javascript
function foo() {
  try {
    throw new Error('foo');
  } finally {
    return;
  }
}
```
the exception won't be thrown and execution will continue normally. If you remove the `return` from the `finally` block, execution will continue to the end of the `finally` block at which point the exception will be thrown.